src/qiboml/models/abstract.py

```diff
@@ -0,0 +1,67 @@
+"""Defines the general structure of a qiboml module"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from qibo import Circuit
+from qibo.config import raise_error
+from qibo.gates import abstract
+
+from qiboml.backends import TensorflowBackend as JaxBackend
+
+
+@dataclass
+class QuantumCircuitLayer(ABC):
+
+    nqubits: int
+    qubits: list[int] = None
+    circuit: Circuit = None
+    initial_state: "ndarray" = None
```

Comment:
Yeah, no, this was actually just a place holder to indicate a general `ndarray` type, either from `numpy`, `torch`, `tf` or `jax`.

Reply:
I could try to set up a workflow with `mypy` then, should I add it to the `qiboteam/workflows` repo and then load it here from there?

Reply:
Worfklows could be annoying. Workflows in another repo even more.

I'd suggest going in steps: do it here, and then we move to the `qiboteam/workflows`.

Notice that:
- Qibosoq already has it https://github.com/qiboteam/qibosoq/blob/main/.github/workflows/mypy.yml
- there are already issues for that in `qiboteam/workflows`, I've simply never taken the time of implementing because there would have been no one using it
  - https://github.com/qiboteam/workflows/issues/31
  - https://github.com/qiboteam/workflows/issues/32 (`symilar` is part of Pylint, might be needed only when we'll remove Pylint)

Reply:
Ok I created a `qiboml.ndarray` type that groups the array types of all the installed frameworks: numpy (and jax) is always there, whereas `torch.Tensor` and `tf.Tensor` are present only when installed.